### ISSUE TITLE
Feature Request: pack to output stream

### ISSUE THREAD

**Issue by marcpawl** [zeroturnaround/zt-zip#76]
Use case:  pack entire directory to an out stream so the file system is not needed to create a temporary zip file. 

I would like for there to be a method that takes a directory as an input parameter and returns a ZipOutputStream.

**Comment by nemecec** [comment]
I created a pull request that would add such a method. You can check the code in the branch if this is what you had in mind?

**Comment by nemecec** [comment]
Actually, I found that we already have a method to achieve what you are asking for `ZipUtil.pack(ZipEntrySource[] entries, OutputStream os)`. I have to admit, the API of that method is quite non-intuitive and cumbersome for your use-case -- you would need manually iterate over all the files that you want to add to the archive and create a `FileSource` object for each file. And then pass that array of `FileSource` objects to the previously mentioned method.

Long story short: I think it is still worthwhile to add that method (as in my pull request), taking a directory as input.

**Comment by marcpawl** [comment]
Yes, that looks much nicer.
Thanks

**Comment by nemecec** [comment]
Ok, now I added also `ZipUtil.pack(File sourceDir, OutputStream os)` method (with some overloaded versions, for more control over the process).
